### DISCUSSION TITLE
How to roll back an upgrade?

### DISCUSSION THREAD

**Discussion by Korijn** [linkerd/linkerd2#7910] [Q&A]
Just wondering how I am supposed to rollback an upgrade in case things go awry?

**Comment by alpeb** [comment]
If you're relying on Helm (as recommended for production), `helm rollback` should bring you back to the previous release you had, using the same settings. If you had rolled out the dataplane already to bring the proxies into the new version, you'll have to roll them out again to sync their version again with the rolled back version. Needless to say, it's a good idea to always try any production changes in an staging environment first.
